### PR TITLE
PrgEnv-pgi support

### PR DESCRIPTION
**Pull request by jgphpc**
PrgEnv-pgi support
- depends on https://github.com/hpcugent/easybuild-framework/pull/1729

### CODE REVIEW

File: easybuild/easyconfigs/c/CrayPGI/CrayPGI-2016.04.eb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+homepage = 'http://www.pgroup.com/'
+description = """Toolchain using Cray compiler wrapper, using PrgEnv-pgi module."""
+
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+
+dependencies = [
+     # PrgEnv version is not pinned, as Cray recommends to use the latest (default) version
+     # ('PrgEnv-pgi/' + version, EXTERNAL_MODULE),
+     ('PrgEnv-pgi/5.2.82', EXTERNAL_MODULE),
```
Comment (boegel) [inline easybuild/easyconfigs/c/CrayPGI/CrayPGI-2016.04.eb]: @jgphpc this should *not* be version pinned, see comment, just use `'PrgEnv-pgi'`

File: easybuild/easyconfigs/c/CrayPGI/CrayPGI-2016.04.eb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+     ('pgi/16.3.0' , EXTERNAL_MODULE),
+     ('cray-mpich/7.3.2', EXTERNAL_MODULE),
+     # ('cray-libsci/16.03.1', EXTERNAL_MODULE), # Cray does not provide cray-libsci for PrgEnv-pgi
```
Comment (boegel) [inline easybuild/easyconfigs/c/CrayPGI/CrayPGI-2016.04.eb]: hmm, this is going to yield problems whenever BLAS/LAPACK is required?

@gppezzi

File: easybuild/easyconfigs/c/CrayPGI/CrayPGI-2016.04.eb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+]
+
+moduleclass = 'toolchain'
+
```
Comment (boegel) [inline easybuild/easyconfigs/c/CrayPGI/CrayPGI-2016.04.eb]: drop empty line please ^_^

**Comment by boegel** [comment]
Jenkins: ok to test

@jgphpc I guess using HPL here as a testcase for this toolchain doesn't make much sense, since there's no BLAS/LAPACK support for CrayPGI (at least not as is)?

**Comment by jgphpc** [comment]
@boegel yes, with intel mkl: https://www.pgroup.com/support/link.htm#mkl

**Comment by boegel** [comment]
@jgphpc So, should an Intel MKL module provided by Cray be part of `CrayPGI` then?

**Comment by jgphpc** [comment]
@boegel it's possible to link without mkl, can you share you hpl testcase ?

###### without intel mkl
*  ftn pgi_mm.F -lblas -dynamic
```
# libblas.so.0 => /opt/pgi/16.3.0/linux86-64/16.3/lib/libblas.so.0
```

###### with intel mkl
* ftn  -L$MKLROOT/lib/intel64 -lmkl_intel_lp64 -lmkl_intel_thread -lmkl_core -pgf90libs -liomp5 -lpthread -lm -L$MKLROOT/../compiler/lib/intel64 pgi_mm.F -dynamic
* ldd a.out  |grep mkl
```
	libmkl_intel_lp64.so => /opt/intel/compilers_and_libraries_2016.1.150/linux/mkl/lib/intel64/libmkl_intel_lp64.so (0x00007f30c8894000)
	libmkl_intel_thread.so => /opt/intel/compilers_and_libraries_2016.1.150/linux/mkl/lib/intel64/libmkl_intel_thread.so (0x00007f30c72b9000)
	libmkl_core.so => /opt/intel/compilers_and_libraries_2016.1.150/linux/mkl/lib/intel64/libmkl_core.so (0x00007f30c5982000)
```

**Comment by boegel** [comment]
@jgphpc nothing special about HPL, just regular easyconfigs, cfr. https://github.com/hpcugent/easybuild-easyconfigs/tree/master/easybuild/easyconfigs/h/HPL

But, to use the BLAS readily provided by the `pgi` module, you may need to fiddle with the framework support a bit, to get things like `$LIBBLAS` defined, see https://github.com/hpcugent/easybuild-framework/tree/master/easybuild/toolchains/linalg, in particular `libsci.py`

**Comment by boegel** [comment]
(open/close to trigger Travis, need to look into a better way but this is what's recommended by Travis support...)

**Comment by boegel** [comment]
Good to go now https://github.com/hpcugent/easybuild-framework/pull/1729 is merged, thanks @jgphpc!

**Pull request merged**